src/contract.ts

```diff
@@ -25,24 +50,64 @@ function parseFelt(candidate: string): BN {
   }
 }
 
+function buildCall(contract: Contract, functionAbi: FunctionAbi): AsyncContractFunction {
+  return async function (args: Args, blockIdentifier: BlockIdentifier): Promise<any> {
+    return contract.call(functionAbi.name, args, blockIdentifier);
+  };
+}
+
+function buildInvoke(contract: Contract, functionAbi: FunctionAbi): AsyncContractFunction {
+  return async function (args: Args): Promise<any> {
```

Comment:
`Args` is still the v2 way of accepting invocation/call arguments. I think we should support multiple formats, like ethers.js does and as described in https://github.com/0xs34n/starknet.js/discussions/130. wdyt?
Same for return values, which are objects only right now

Reply:
we agreed to improve on this in the future and keep one single format for now.